server/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -47,6 +54,57 @@
      */
     public static final long UNAVAILABLE_EXPECTED_SHARD_SIZE = -1;
 
+    private static final ParseField STATE = new ParseField("state");
+    private static final ParseField PRIMARY = new ParseField("primary");
+    private static final ParseField NODE = new ParseField("node");
+    private static final ParseField RELOCATING_NODE = new ParseField("relocating_node");
+    private static final ParseField SHARD = new ParseField("shard");
+    private static final ParseField INDEX = new ParseField("index");
+    private static final ParseField EXPECTED_SHARD_SIZE_IN_BYTES = new ParseField("expected_shard_size_in_bytes");
+    private static final ParseField RECOVERY_SOURCE = new ParseField("recovery_source");
+    private static final ParseField ALLOCATION_ID = new ParseField("allocation_id");
+    private static final ParseField UNASSIGNED_INFO = new ParseField("unassigned_info");
+
+    private static final ConstructingObjectParser<ShardRouting, String> PARSER =
```

Comment:
I think we need to have a discussion on whether we want to keep sending this to a user via the HLRC at all, I think the best option is not to send it, but that would require more work for us. We should discuss this